processing/src/main/java/org/apache/druid/query/aggregation/post/FinalizingFieldAccessPostAggregator.java

```diff
@@ -26,16 +26,20 @@
 import org.apache.druid.query.aggregation.AggregatorFactory;
 import org.apache.druid.query.aggregation.PostAggregator;
 import org.apache.druid.query.cache.CacheKeyBuilder;
+import org.apache.druid.segment.column.ValueType;
+import org.apache.druid.segment.column.ValueTypes;
 
 import java.util.Comparator;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 import java.util.function.Function;
 
 public class FinalizingFieldAccessPostAggregator implements PostAggregator
 {
   private final String name;
   private final String fieldName;
+  private final ValueType finalizedType;
```

Comment:
Hmm, I'm undecided if it should it be in equals and hashCode... Should calling `decorate` still be equal to the postagg before calling decorate? If so, then I think it should be ignored in these places since the value is computed.

Reply:
It seems ok to omit, looking at the code, the only time post-aggs exist as undecorated is for a short period in the constructors of the queries which can have post-aggregators.

Reply:
Nice finding. Would you leave a comment about why the `finalizedType` is not in those methods? It could be confusing otherwise.